src/mapblock.cpp

```diff
@@ -259,25 +321,20 @@ static void correctBlockNodeIds(const NameIdMapping *nimap, MapNode *nodes,
 	std::unordered_set<content_t> unnamed_contents;
 	std::unordered_set<std::string> unallocatable_contents;
 
-	bool previous_exists = false;
-	content_t previous_local_id = CONTENT_IGNORE;
-	content_t previous_global_id = CONTENT_IGNORE;
+	// Used to cache local to global id lookup.
+	IdIdMapping mapping_cache;
 
 	for (u32 i = 0; i < MapBlock::nodecount; i++) {
 		content_t local_id = nodes[i].getContent();
-		// If previous node local_id was found and same than before, don't lookup maps
-		// apply directly previous resolved id
-		// This permits to massively improve loading performance when nodes are similar
-		// example: default:air, default:stone are massively present
-		if (previous_exists && local_id == previous_local_id) {
-			nodes[i].setContent(previous_global_id);
+
+		if (mapping_cache.get(local_id) != 0xFFFF) {
+			nodes[i].setContent(mapping_cache.get(local_id));
```

Comment:
```suggestion
		if (auto found = mapping_cache.get(local_id); found != 0xFFFF) {
			nodes[i].setContent(found); 
```

Reply:
Thanks, done.

(The compiler did optimize the additional mem access away, but not relying on it is nice.)